File: ZTW2/CarMarket/src/CarMarket/Entities/Offer.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace CarMarket.Entities
{
    public class Offer
    {
        public int OfferId { get; set; }

        public int? UserId { get; set; }
        public User User { get; set; }

        public string Model { get; set; }
        public int Year { get; set; }
        public int Mileage { get; set; }
        public string Description { get; set; }
        public double Price { get; set; }
        public string Color { get; set; }
        public string Fuel { get; set; }
        public string Category { get; set; }
        public int SeatsNb { get; set; }
        public int DoorsNb { get; set; }
        public int Displacement { get; set; }
        public string Gearbox { get; set; }
        public string Drive { get; set; }
        public bool Damaged { get; set; }
        public bool ABS { get; set; }
        public bool Airbags { get; set; }
        public bool CentralLock { get; set; }
        public bool AirCond { get; set; }
        public DateTime StartTime { get; set; }
        public DateTime EndTime { get; set; }
        public string Status { get; set; }

        public Album Album { get; set;  }
        public int AlbumID { get; set; }

    }
}

File: ZTW2/CarMarket/src/CarMarket/wwwroot/app/core/domain/offer.ts
export class Offer {
    Id: number;
    UserID: number;
    Model: string;
    Year: number;
    Mileage: number;
    Description: string;
    Price: number;
    Color: string;
    Fuel: string;
    Category: string;
    SeatsNb: number;
    DoorsNb: number;
    Displacement: number;
    Gearbox: string;
    Drive: string;
    Damaged: boolean;
    ABS: boolean;
    Airbags: boolean;
    CentralLock: boolean;
    AirCond: boolean;
    StartTime: Date;
    EndTime: Date;
    Status: string;
    Make: string;

    constructor(id: number,
        userID: number,
        model: string,
        year: number,
        mileage: number,
        description: string,
        price: number,
        color: string,
        fuel: string,
        category: string,
        seatsNb: number,
        doorsNb: number,
        displacement: number,
        gearbox: string,
        drive: string,
        damaged: boolean,
        abs: boolean,
        airbags: boolean,
        centrallock: boolean,
        aircond: boolean,
        starttime: Date,
        endtime: Date,
        status: string,
        make: string) {
        this.Id = id;
        this.UserID = userID;
        this.Model = model;
        this.Year = year;
        this.Mileage = mileage;
        this.Description = description;
        this.Price = price;
        this.Color = color;
        this.Fuel = fuel;
        this.Category = category;
        this.SeatsNb = seatsNb;
        this.DoorsNb = doorsNb;
        this.Displacement = displacement;
        this.Gearbox = gearbox;
        this.Drive = drive;
        this.Damaged = damaged;
        this.ABS = abs;
        this.Airbags = airbags;
        this.CentralLock = centrallock;
        this.AirCond = aircond;
        this.StartTime = starttime;
        this.EndTime = endtime;
        this.Status = status;
        this.Make = make;
    }



}